src/Docker.DotNet/Endpoints/ImageOperations.cs

```diff
@@ -164,13 +178,17 @@ public Task<Stream> PushImageAsync(string name, ImagePushParameters parameters,
             return this._client.MakeRequestForStreamAsync(this._client.NoErrorHandlers, HttpMethod.Post, $"images/{name}/push", queryParameters, null, RegistryAuthHeaders(authConfig), CancellationToken.None);
         }
 
-        public Task PushImageAsync(string name, ImagePushParameters parameters, AuthConfig authConfig, IProgress<JSONMessage> progress, CancellationToken cancellationToken = default(CancellationToken))
+        private Task<Stream> LoadImageAsyncPrivate(ImageLoadParameters parameters, Stream imageStream, IProgress<JSONMessage> progress, CancellationToken cancellationToken)
```

Comment:
You dont actually need these private methods. Just do the validation in the public one and pass this._client.MakeRequestForStreamAsync to the Monitor call.